tools/bin/triton-scry

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+#
+# Copyright 2023 MNX Cloud, Inc.
+#
+
+#
+# Use tmate to create a shared ssh session that support personnel can connect
+# to.
+#
+# Usage:
+#     triton-scry
+#
+
+if [[ -n "$TRACE" ]]; then
+    export PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
+    set -o xtrace
+fi
+
+# Ensure our path includes pkgsrc, even if it wouldn't otherwise.
+PATH=/usr/bin:/usr/sbin:/smartdc/bin:/opt/smartdc/bin:/opt/local/bin:/opt/local/sbin:/opt/tools/bin:/opt/tools/sbin:/opt/smartdc/agents/bin
+
+# We only need to get consent once
+consent=0
+function get_consent {
+    if (( consent == 0 )); then
+        m+='Press ^C to cancel or any other key to continue.'
+        printf '%b' "$m"
+        read -r
+        consent=1
+    fi
+}
+
+function ensure_pkgsrc {
+    if ! command -V pkgin >/dev/null 2>&1 ; then
+        printf 'The pkgsrc package manager must be installed in the global zone\n'
+        get_consent
+        /smartdc/bin/pkgsrc-setup
+    fi
+}
+
+function ensure_tmate {
+    if ! command -V tmate >/dev/null 2>&1 ; then
+        ensure_pkgsrc
+        printf 'The necessary packages will now be installed.\n'
+        get_consent
+        pkgin -y install tmate
+    fi
+}
+
+function usage {
+    printf 'Usage: %s [-S <path>]\n\n' "$(basename "$0")"
+    printf 'Basic options:\n'
+    printf ' -S <path>\tattach to an existing scry session socket <path>\n'
+    exit "$1"
+}
+
+function cleanup {
+    rm "${conf:?}"
```

Comment:
I always forget that `${var:?}` aborts the whole invocation if `var` is not set.

Reply:
It does, it returns an error if it's not set. In this case, it shouldn't make a difference unless something is modifying the program in memory behind our back.

I generally use that with rm just out of habit to protect against things like this:

    rm -rf "$dir/$file"

Where if both are unset, it will evaluate to `rm -rf /`.